mercat/cli/common/src/harness.rs

```diff
@@ -98,12 +117,27 @@ impl TryFrom<(u32, String)> for Transfer {
             reason: format!("Pattern did not match {}", segment),
         })?;
         let ticker = caps[3].to_string().to_uppercase();
+        let auditors = if caps.get(9) == None {
+            vec![]
+        } else {
+            caps[9]
+                .to_string()
+                .to_lowercase()
+                .split(",")
+                .map(|name| Party::try_from((name, PartyKind::Auditor)))
+                .collect::<Result<_, _>>()?
+        };
+        let tx_name = if caps.get(11) == None {
+            None
+        } else {
+            Some(caps[11].trim().to_string())
+        };
```

Comment:
If you enable the **case-insensitive flag** `(?i)` in your regex expression, you can avoid the allocation of each `String` in `cap[9]` parse, and you could use directly `&str` (0 deep-copies).


```suggestion
        let auditors = caps.get(9).map( |c| {
            c.split(",")
                .map(|n| Party::try_from((n, PartyKind::Auditor)))
                .collect::<Result<_, _>>()
        }).unwrap_or(Ok(vec![]))?;
        
        let tx_name = caps.get(11).map(|c| c.as_str().trim().to_string());
```

Reply:
This did not work, we can chat on Monday.